cosmo/graphqlclient.py

```diff
@@ -1,10 +1,44 @@
 import json
 from string import Template
+from urllib.parse import urlencode
 
 import requests
 
 
 class GraphqlClient:
+    def __init__(self, url, token):
+        self.url = url
+        self.token = token
+
+        v = self.query_version()
+
+        if v == "3.x":
+            self.child_client = GraphqlClientV3(url, token)
+        elif v == "4.x":
+            self.child_client = GraphqlClientV4(url, token)
+        else:
+            raise Exception("Unknown Version")
+
+    def query_version(self):
+        r = requests.get(
+            f"{self.url}/api/status/",
+            headers={
+                "Authorization": f"Token {self.token}",
+                "Content-Type": "application/json",
+                "Accept": "application/json",
+            },
+        )
+        if r.status_code != 200:
+            raise Exception("Error querying api: " + r.text)
+
+        json = r.json()
+        return "3.x" if str(json['netbox-version']).startswith("wc_3") else "4.x"
```

Comment:
couldn't we simplify this? personnaly I find the double version check there and in ```__init__``` to be a bit redundant

Reply:
I removed the redundant check and initialze the correct strategy based on the version field in `__init__.py`